test/test_examples.py

```diff
@@ -59,20 +61,22 @@ def test_some_headers(self):
 
 
 class TestProcessExamples:
```

Comment:
Think it would be useful to test Examples when the inference function is a coroutine?

Reply:
Yup sounds good, I'll add a test